### PR TITLE
even more modular...

### PR DESCRIPTION
**Pull request by mcy**
tl;dr variables now live in the `responses.json` too and features/moods are now decoupled, so introducing a mood is as simple as modifying `responses.json`.

Also all of mommy's error messages are now in-character and use variables as appropriate.

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -3,61 +3,170 @@
...
-            ];"#
+            r"
+            const CONFIG: Responses<'static> = Responses {{
+                vars: &[{vars}],
+                moods: &[{responses}],
+            }};
+            const MOOD: &'static Var<'static> = &CONFIG.vars[0];
+            const EMOTE: &'static Var<'static> = &CONFIG.vars[1];
+            const PRONOUN: &'static Var<'static> = &CONFIG.vars[2];
+            const ROLE: &'static Var<'static> = &CONFIG.vars[3];
```
Comment (workingjubilee) [inline build.rs:168]: This declares a `const ROLE: &'static Var<'static>`

File: src/main.rs [hunk 2]
```diff
@@ -69,130 +67,123 @@ fn select_response(response_type: ResponseType) -> Result<String, String> {
...
-        output.push_str(&input[last_position..index]);
-        output.push_str(&options[rng.usize(..options.len())]);
-        last_position = index + matched.len();
+impl Var<'_> {
+    fn load(&self, rng: &Rng) -> Result<String, String> {
+        self.maybe_load(rng).ok_or_else(|| {
+            let role = if std::ptr::eq(self, ROLE) {
+                "mommy(?)".to_string()
+            } else {
+                match ROLE.load(rng) {
```
Comment (workingjubilee) [inline src/main.rs:155]: However, this performs a load by relying on pointer equality.

I recently woke up, so I am not 100% awake for the "Rust reference semantics" part of my brain, but this logic is at risk of incorrect reasoning as `ROLE` is a `const`, allowing its de facto duplication and thus spurious pointer inequality.

Even if we could use a `const`, other little ones trying to help mommy might be easily confused, and we wouldn't want that, would we~?

File: build.rs (hunk 1 above)
Comment (workingjubilee) [inline build.rs:168]: For the preceding reasons, to guarantee these all have one location in memory, so we can reason about their placement, mommy should use `static` instead. Mommy's a bit fluffy and gets a bit of friction, I guess, huh~? Watch out for zaps:zap: 
```suggestion
            static CONFIG: Responses<'static> = Responses {{
                vars: &[{vars}],
                moods: &[{responses}],
            }};
            static MOOD: &'static Var<'static> = &CONFIG.vars[0];
            static EMOTE: &'static Var<'static> = &CONFIG.vars[1];
            static PRONOUN: &'static Var<'static> = &CONFIG.vars[2];
            static ROLE: &'static Var<'static> = &CONFIG.vars[3];
```

**Review by Gankra — APPROVED** [review]
needs rebase but seems like a reasonable change, in that the bulk of other changes now become largely data and not code

would love more comments describing wtf is going on, since the code is kinda A Lot now

i'm gonna trust you and jubilee to hash out the pointer thing because i am kinda loopy rn

File: build.rs (hunk 1 above)
Comment (mcy) [reply build.rs]: > Watch out for zaps⚡
my brain: what the fuck does this have to do with pointer zapping 😱

I think I'm going to just. not use pointer equality. This doesn't actually matter (as far as perf goes) at the LLVM level, at the end of the day...

Wait, why am i thinking about the perf of a novelty D/s build system plugin?

**Pull request merged**